SharedCode/src/commonMain/kotlin/ApplicationContract.kt

```diff
@@ -4,14 +4,14 @@ import kotlinx.coroutines.CoroutineScope
 
 interface ApplicationContract {
     interface View {
-        fun setStationNames(stationNames: List<String>)
+        fun setStationNames(stationNames: List<StationDetails>)
         fun createAlert(alertMessage: String, alertTitle: String)
         fun setTableData(data: List<DepartureInformation>)
     }
 
     abstract class Presenter : CoroutineScope {
         abstract fun onViewTaken(view: View)
-        abstract fun makeTrainSearch(originCrs: String, destinationCrs: String, dateTime: String)
+        abstract fun makeTrainSearch(originCrs: String, destinationCrs: String, dateTime: String, adultCount: String, childCount: String)
         abstract fun formatDateTimeInput(input: String, format: String): String
```

Comment:
I have a hunch that this processing should be done in the common core right away and this function should not be in the contract. as in I feel like formatting should not be exposed to the view and sending the same data back and forth multiple times is sus.